src/extends/creep/actions.js

```diff
@@ -42,6 +42,10 @@ Creep.prototype.recharge = function () {
   }
 
   // As a last resort harvest energy from the active sources.
+  if (this.getActiveBodyparts(WORK) <= 0) {
+    // Still returning true because the creep still does need to recharge.
+    return true
+  }
   const sources = this.room.find(FIND_SOURCES_ACTIVE)
```

Comment:
The comment on Line 44 should be above Line 49, no?

Reply:
No, the body check is part of the source harvesting block.